sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/MonotonicallyIncreasingID.scala

```diff
@@ -40,13 +42,23 @@ import org.apache.spark.sql.types.{DataType, LongType}
       represent the record number within each partition. The assumption is that the data frame has
       less than 1 billion partitions, and each partition has less than 8 billion records.""",
   extended = "> SELECT _FUNC_();\n 0")
-case class MonotonicallyIncreasingID() extends LeafExpression with Nondeterministic {
+case class MonotonicallyIncreasingID(offset: Long = 0) extends LeafExpression
+  with Nondeterministic with Logging {
+
+  def this() = {
+    this(offset = 0)
+  }
+
+  def this(offset: Expression) = {
+    this(offset = MonotonicallyIncreasingID.parseExpression(offset))
+    logDebug(s"offset is $offset")
+  }
 
   /**
-   * Record ID within each partition. By being transient, count's value is reset to 0 every time
-   * we serialize and deserialize and initialize it.
+   * Record ID within each partition. By being transient, count's value is reset to offset every
+   * time we serialize and deserialize and initialize it.
    */
-  @transient private[this] var count: Long = _
+  @transient private[this] var count: Long = offset
```

Comment:
This probably does not work. You need to set the value in the `MonotonicallyIncreasingID.initInternal()`. This is why your tests are failing.